core/src/client/async_client/mod.rs

```diff
@@ -633,10 +637,21 @@ async fn background_task<S, R>(
 	});
 	futures_util::pin_mut!(backend_event);
 
+	let frontend_event = futures_util::stream::unfold(frontend, |mut frontend| async {
+		let res = frontend.recv().await;
+		if let Some(res) = res {
+			Some((res, frontend))
+		} else {
+			None
+		}
+	});
+	futures_util::pin_mut!(frontend_event);
```

Comment:
Just curious; so this crates a `Stream` from `frontend`. Why do we need it though? Is it not possible to use `frontend.recv()` instead of `frontend.next()` below?